staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates/util.go

```diff
@@ -46,7 +46,7 @@ func GetHumanCertDetail(certificate *x509.Certificate) string {
 		usages = append(usages, fmt.Sprintf("%d", curr))
 	}
 
-	validServingNames := []string{}
+	validServingNames := make([]string, 0, len(certificate.IPAddresses))
```

Comment:
This seems like both a premature optimization and actually incorrect, since we know at least one more will be appended.

Reply:
(which is also why this typically isn't done unless the code is very performance sensitive, it's very easy to miss the mark on this)